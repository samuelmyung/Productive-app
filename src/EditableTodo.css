@-webkit-keyframes fadeOut {
  0% {opacity: 1}
  100% {opacity: 0}
}

@keyframes fadeOut {
  0% {opacity: 1}
  100% {opacity: 0}
}

.deletedTodo {
  animation: fadeOut .5s forwards;
}