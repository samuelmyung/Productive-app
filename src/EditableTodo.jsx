import React, { useState } from "react";
import Todo from "./Todo";
import TodoForm from "./TodoForm";

/** Show editable todo item.
 *
 * Props
 * - todo
 * - update(): fn to call to update a todo
 * - remove(): fn to call to remove a todo
 *
 * State
 * - isEditing
 *
 * EditableTodoList -> EditableTodo -> { Todo, TodoForm }
 */

function EditableTodo({todo, update, remove}) {
  const [isEditing, setIsEditing] = useState(false);

  /** Toggle if this is being edited */
  function toggleEdit() {
    setIsEditing(isEditing => !isEditing);
  }

  /** Call remove fn passed to this. */
  function handleDelete() {
    remove(todo.id);
  }

  /** Edit form saved; toggle isEditing and update in ancestor. */
  function handleSave(formData) {
    update(formData);
    toggleEdit();
  }

  return (
      <div className="EditableTodo">

                {isEditing && <TodoForm handleSave={handleSave} initialFormData={todo}  />}

                {!isEditing && <div className="mb-3">
                  <div className="float-end text-sm-end">
                    <button
                        className="EditableTodo-toggle btn-link btn btn-sm"
                        onClick={toggleEdit}>
                      Edit
                    </button>
                    <button
                        className="EditableTodo-delBtn btn-link btn btn-sm text-danger"
                        onClick={handleDelete}>
                      Del
                    </button>
                  </div>
                  <Todo
                    id={todo.id}
                    title={todo.title}
                    description={todo.description}
                    priority={todo.priority}
                  />
                </div>}

      </div>
  );
}

export default EditableTodo;
