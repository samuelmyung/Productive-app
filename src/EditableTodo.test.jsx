import { describe, it, expect } from "vitest";

import React from "react";
import { render, fireEvent, wait } from "@testing-library/react";
import EditableTodo from "./EditableTodo";

const TODO = {
    id: 1,
    title: 'test title',
    description: 'test description',
    priority: 1
}

it('renders without crashing', function () {
    render(<EditableTodo todo={TODO} />);
});

it('matches snapshot with initial todos', function () {
    const {container} = render(
    <EditableTodo todo={TODO} />
    );
    expect(container).toMatchSnapshot();
});