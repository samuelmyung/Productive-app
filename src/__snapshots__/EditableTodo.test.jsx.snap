// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot with initial todos 1`] = `
<div>
  <div
    class="EditableTodo"
  >
    <div
      class="mb-3"
    >
      <div
        class="float-end text-sm-end"
      >
        <button
          class="EditableTodo-toggle btn-link btn btn-sm"
        >
          Edit
        </button>
        <button
          class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
        >
          Del
        </button>
      </div>
      <div
        class="Todo"
      >
        <div>
          <b>
            test title
          </b>
           
          <small>
            (priority: 
            1
            )
          </small>
        </div>
        <div>
          <small>
            test description
          </small>
        </div>
      </div>
    </div>
  </div>
</div>
`;
